File: Kısım2C/src/Kisi.c
#include "Kisi.h"
#include "Telefon.h"
#include "KimlikNo.h"
Kisi KisiOlustur()
{
    Kisi this=(Kisi)malloc(sizeof(struct KISI));
    this->kimlikNo=KimlikNoOlustur();
    this->telefon=telefonOlustur();
    this->rastgele=RastgeleOlustur();
    this->kisiUret=&_kisiUret;
    this->KisiYoket=&_KisiYoket;
	return this;
}
int _dosyaSatirSayisiBul(const Kisi this) //isimler dosyasının satır sayısı bulunuyor
{
		FILE *dosya;
		this->satirSayisi=1;
		char oku;

		dosya=fopen("isimler.txt", "r");
		while(!feof(dosya)){
			oku=getc(dosya);
			if(oku=='\n'){
				this->satirSayisi++;
			}
		}
		fclose(dosya);
		return this->satirSayisi;
}

char* _rastgeleAdSoyadGetir(const Kisi this)
{
    int rastgeleSatir=0;
    int sayac=0;
int dosyaDegerSayisi=_dosyaSatirSayisiBul(this); //isimler dosyasında kaç adet satır olduğu bulunuyor ve rastgele gelen değerlerin bu satır sayısı aralığında olması sağlanıyor
    char* isim="";
    while(rastgeleSatir==0)
        rastgeleSatir=this->rastgele->randomDegerGetir(this->rastgele)%dosyaDegerSayisi;
    FILE * fp;
    char * line = NULL;
    size_t len = 0;

    fp = fopen("isimler.txt", "r");
    if (fp == NULL)
        exit(EXIT_FAILURE);

    while ((getline(&line, &len, fp)) != -1) {
        sayac++;
        if(sayac==rastgeleSatir)
        {
            strtok(line,"\n");
            this->adSoyad=line;
            break;
        }
    }
    fclose(fp);
    return this->adSoyad;
}
void _kisileriDosyayaYaz(const Kisi this)
{

    FILE *fp=fopen("Kisiler.txt","a");

    fprintf(fp,"%s %s %d %s\n",this->tcNo,this->adSoyad,this->yas,this->kisitelefonVeImei);
fclose(fp);
}

void _kisiUret(const Kisi this)
{
    this->tcNo=this->kimlikNo->tcKimlikNoGetir(this->kimlikNo);
    this->kisitelefonVeImei=this->telefon->telefonVeImeiGetir(this->telefon);
    this->yas=this->rastgele->randomDegerGetir(this->rastgele)%100;
    this->adSoyad=_rastgeleAdSoyadGetir(this);
    _kisileriDosyayaYaz(this);
}
void _KisiYoket(Kisi this)
{
    if(this==NULL)return;
    this->kimlikNo->KimlikNoYoket(this->kimlikNo);
    this->telefon->TelefonYoket(this->telefon);
    this->rastgele->RastgeleYoket(this->rastgele);
    free(this);


}

File: Kısım1Java/Kutuphane/RASTGELEKISIURET/src/MyPackage/Kisi.java

package MyPackage;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Scanner;

public class Kisi {
    private final String dosyaYolu;
    private String telefonVeImeiNumarasi;
    private String tcNo;  
    private  KimlikNo tc;
    private  String adSoyad;
    private  int yas;
    private  Telefon telefon;  //imei ve telefon birlikte geliyor
    private String bilgiler;
  Rastgele ran;
  Rastgele rastgeleAdSoyad;
 ArrayList<String> kisiler;
    Kisi() throws FileNotFoundException
    {
       tc=new KimlikNo();
       telefon=new Telefon();
       ran=new Rastgele(100);
       kisiler=new ArrayList<String>();
       dosyaYolu="isimler.txt";
       rastgeleAdSoyad=new Rastgele(dosyaSatirSayisiBul());
       bilgiler="";
    }
    public void kisiUret() throws FileNotFoundException, IOException
    {
        tcNo= tc.tcKimlikNoGetir();
        yas=(int)ran.randomDegerGetir();
        telefonVeImeiNumarasi=telefon.telefonVeImeiGetir();
        adSoyad=rastgeleAdSoyadGetir();  
      
        bilgiler=tcNo+" "+adSoyad+yas+" "+telefonVeImeiNumarasi;
        kisiler.add(bilgiler);
    }
    public void kisileriDosyayaYaz() throws IOException
    {
         File file = new File("Kisiler.txt");
        if (!file.exists()) {
            file.createNewFile();
        }

        FileWriter fileWriter = new FileWriter(file,false);
        BufferedWriter bWriter = new BufferedWriter(fileWriter);

        for(String kisi:kisiler)
        {
               bWriter.write(kisi.toString());
                        bWriter.newLine();
        }
        bWriter.close();
    }
    private int dosyaSatirSayisiBul() throws FileNotFoundException //isimler dosyasının satır sayısı bulunuyor
    {
         java.io.File dosyaSatirSayisi = new java.io.File(dosyaYolu);
          Scanner satir = new Scanner(dosyaSatirSayisi);
          int satirSayisi=0;
  

          while (satir.hasNext())  //her satır bitiminde sayac bir arttırılıyor
          {
              satirSayisi++;                 
              satir.nextLine();
          }    
          return satirSayisi+1;
    }
    private String rastgeleAdSoyadGetir() throws FileNotFoundException
    {
            int rastgeleSatir=0;
            int sayac=0;
            String isim="";

            while(rastgeleSatir==0)
                    rastgeleSatir=(int)rastgeleAdSoyad.randomDegerGetir();
   
           java.io.File dosyakOku = new java.io.File("isimler.txt");
          Scanner input = new Scanner(dosyakOku);
          while (input.hasNext())  
          {
              sayac++;
              if(sayac==rastgeleSatir)        
            isim = input.nextLine()+ " ";
              else
                  input.nextLine();
                    
          }
          return isim;
    }
  }
    
    

